mmv1/templates/terraform/constants/dataplex_entry.go.tmpl

```diff
@@ -0,0 +1,127 @@
+// GetEntry supports up to 100 aspects. Therefore we set a threshold at 99.
+const maxAspectNumber = 99
+
+// NumberOfAspectsValidation checks if the number of aspects on an entry exceeds certain threshold.
+func NumberOfAspectsValidation(i interface{}, k string) (warnings []string, errors []error) {
+	s, isSlice := i.([]interface{})
+	m, isMap := i.(map[string]interface{})
+
+	if !isSlice && !isMap {
+		errors = append(errors, fmt.Errorf("expected type of field %q to be array, but got %T", k, i))
+		return warnings, errors
+	}
+
+	if len(s)+len(m) > maxAspectNumber {
+		errors = append(errors, fmt.Errorf(
+			"field %q has an invalid content: %q. The maximal number of aspects is 99.",
+			k, i,
+		))
+	}
+
+	return warnings, errors
+}
+
+// ProjectNumberValidation checks if the input string conforms to the pattern:
+// "projects/<project-number>/<anything>"
+func ProjectNumberValidation(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of field %q to be string, but got %T", k, i))
+		return warnings, errors
+	}
+
+	var projectNumberRegex = regexp.MustCompile(`^projects\/[1-9]\d*\/.+$`)
+	if !projectNumberRegex.MatchString(v) {
+		errors = append(errors, fmt.Errorf(
+			"field %q has an invalid format: %q. Expected format: 'projects/<project-number>/<anything>'. Please note that project IDs are not supported.",
+			k, v,
+		))
+	}
+
+	return warnings, errors
+}
+
+// ProjectNumberValidation checks if the input string conforms to the pattern:
+// "projects/<project-number>/<anything>"
+func AspectProjectNumberValidation(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of field %q to be string, but got %T", k, i))
+		return warnings, errors
+	}
+
+	var numberDotAnythingRegex = regexp.MustCompile(`^[1-9]\d*\..+$`)
+
+	if !numberDotAnythingRegex.MatchString(v) {
+		errors = append(errors, fmt.Errorf(
+			"field %q has an invalid format: %q. Expected format: '<project number>.anything'. Please note that project IDs are not supported.",
+			k, v,
+		))
+	}
+
+	return warnings, errors
+}
+
+func FilterAspects(aspectKeySet map[string]struct{}, res map[string]interface{}) {
+	if res["aspects"] == nil {
+		return
+	}
+	aspectsMap := res["aspects"].(map[string]interface{})
```

Comment:
Can this be updated to return an error rather than panicing? Ditto for the other unchecked type assertions below (I know that TPG has a bunch of these, but it shouldn't :-)